src/shared/lib/table.ts

```diff
@@ -78,3 +79,77 @@ export function useTable<T>(fetchFn: TableFetchFn<T>) {
     setSize,
   };
 }
+
+// for generating blocksListTable in paginated way
+export function useBlockTable() {
```

Comment:
Here are the reasons why I think this composable should be refactored:

- It has duplication of pagination logic in `useTable`
- it has `async function fetch()`, which has no reasons to be async. It calls `await transactionModel.fetchBlocksList`, but this function also has no reasons to be async if you look inside. Since `fetch` is `async`, all other functions also have to be `async`: `setPage`, `setSize` etc.
- `loading` flag is useless, because whenever it is set to `true`, it will be immediately set back to `false`, because there is no any asynchronicity in the actuality.
- the logic of function is kind of messy. Externally, you put data inside through `setTransaction(items)`. Then, in order to get the paginated view of transactions, you call `await fetch()` which takes `selectedTransactions`, applies `pagination.value` to it and sets the value into `items.value`, also updating `pagination.value` for some reason (why?). 
- The name `useBlockTable` suggests that it is responsible for a table of some block. It is located in the `shared/lib/table.ts` next to a generic `useTable` function which works over any kinds of paginated async values. `useBlockTable`, however, belongs to Blocks domain. There are other places in the app structure where such specific logic should be placed.

What I would suggest:

- In order to reduce duplication of reactive logic, try to separate it into a reusable composable
- Whenever you have some data updated, do not manually recompute other data which depends on it. This is exactly what `computed`s and watchers are for. In your case, you manually trigger update of `items` through `fetch()` whenever you update the values `items` depend on (`selectedTransactions`, `pagination`). Instead, you should define `items` as a **`computed`**, which is reactively updated on its own whenever its reactive dependencies change.
- Do not use `async` if you don't have reason for it. `async` is a very useful thing, but it brings redundant complexity if used not properly.

Reply:
acknowledged and committed 👍🏼